### PR TITLE
configure local-up-cluster.sh to handle auth proxies

### PR DESCRIPTION
**Pull request by deads2k**
Builds on https://github.com/kubernetes/kubernetes/pull/36707.  Last commit only.

This adds the ability to have an auth proxy provide user info with secured client-cert to the API server.

@sttts

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36838)

### CODE REVIEW

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -434,7 +442,9 @@ EOF
       --service-cluster-ip-range="${SERVICE_CLUSTER_IP_RANGE}" \
       --feature-gates="${FEATURE_GATES}" \
       --cloud-provider="${CLOUD_PROVIDER}" \
-      --cloud-config="${CLOUD_CONFIG}" \
```
Comment (sttts) [inline hack/local-up-cluster.sh]: Intentional to remove this line?

**Comment by sttts** [comment]
one question, otherwise lgtm

File: hack/local-up-cluster.sh (hunk 1 above)
Comment (deads2k) [inline hack/local-up-cluster.sh]: > sttts 4 hours ago Kubernetes member
> Intentional to remove this line?

Nope.  Will fix.

**Comment by deads2k** [comment]
Adding do-not-merge to ensure prereqs merge first.

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -416,12 +422,22 @@ EOF
...
 
+    # Create auth proxy client ca
+    sudo /bin/bash -e <<EOF
+    rm -f "${CERT_DIR}/auth-proxy-client-ca.crt" "${CERT_DIR}/auth-proxy-client-ca.key"
+    openssl req -x509 -sha256 -new -nodes -days 365 -newkey rsa:2048 -keyout "${CERT_DIR}/auth-proxy-client-ca.key" -out "${CERT_DIR}/auth-proxy-client-ca.crt" -subj "/C=xx/ST=x/L=x/O=x/OU=x/CN=ca/emailAddress=x/"
+    echo '{"signing":{"default":{"expiry":"43800h","usages":["signing","key encipherment","client auth"]}}}' > "${CERT_DIR}/auth-proxy-client-ca-config.json"
+EOF
+    create_client_certkey auth-proxy-client-ca auth-proxy system:auth-proxy
+
+    cat "${CERT_DIR}/client-ca.crt" "${CERT_DIR}/auth-proxy-client-ca.crt" > "${CERT_DIR}/complete-client-ca-bundle.crt"
```
Comment (sttts) [inline hack/local-up-cluster.sh]: ugly name. Why not drop the "complete-" prefix?

And I guess you need a sudo around that?

**Comment by deads2k** [comment]
@k8s-bot unit test this: issue #33461

**Pull request merged**